packages/core/src/scenes/Variables.ts

```diff
@@ -0,0 +1,47 @@
+import type {Scene} from './Scene';
+import {createSignal, SimpleSignal} from '@motion-canvas/core/lib/signals';
+
+export class Variables {
+  private signals: {[key: string]: SimpleSignal<any>} = {};
+
+  public constructor(private readonly scene: Scene) {}
+
+  /**
+   * Get variable signal if exists or create signal if not
+   *
+   * @param name - The name of the variable.
+   */
+  public get(name: string) {
+    const variables = this.scene.project.getVariables();
+    if (!(name in variables)) {
+      this.scene.project.logger.warn(
```

Comment:
I'd consider adding another argument to this method to specify the default value of a variable:
```
public get<T>(name: string, initial: T): SimpleSignal<T> {
```
and return it here instead.